### ISSUE TITLE
BUG: Negative depth loss diluted by mask fraction

### ISSUE THREAD

**Issue by zeinali72** [zeinali72/swe-pinn#51]
## Description

In `src/losses/pde.py`, `compute_neg_h_loss` (lines 57-64) applies the mask to `h_pred` before computing the penalty:

```python
h_pred = h_pred * pde_mask          # masked points become 0
return jnp.mean(jax.nn.relu(-h_pred) ** 2)  # mean over ALL points
```

While masked points correctly contribute 0 penalty, `jnp.mean` divides by the total count (including masked points), diluting the loss.

## Impact

If 50% of points are masked, the effective negative-depth penalty is halved. This creates implicit coupling between domain geometry and loss magnitude.

## Fix

Apply mask after computing per-point losses and use a masked mean:

```python
per_point_loss = jax.nn.relu(-h_pred) ** 2
if pde_mask is not None:
    per_point_loss = per_point_loss * pde_mask
    return jnp.sum(per_point_loss) / jnp.maximum(jnp.sum(pde_mask), 1.0)
return jnp.mean(per_point_loss)
```

## Category
**High** · Pillar: Logic Check
